core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -355,6 +355,13 @@ private[spark] object SerDe {
           writeInt(dos, v.length)
           v.foreach(elem => writeObject(dos, elem))
 
+        // Handle Properties
```

Comment:
It is possible to construct a map for the properties and pass it to a helper function on JVM side. That helper function can convert the map into java.util.Properties. The problem is that if there are multiple Scala API functions taking java.util.Properties as parameters, we have to create as many helper functions as them. So I think we can keep current implementation as is for now.